### PR TITLE
feat: Move LauncherView to App.js

### PR DESCRIPTION
**Pull request by doubleface**
This is needed to allow mespapiers app to control clisk konnectors.

Also needed to convert launcher context to a global context to allow it
to be used in multiple places (CozyWebView, App)

#### Checklist

Before merging this PR, the following things must have been done:

* [ ] Faithful integration of the mockups at all screen sizes
* [ ] Tested on iOS
* [X] Tested on Android
* [ ] Localized in English and French
* [ ] All changes have test coverage
* [ ] Updated README & CHANGELOG, if necessary

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -48,6 +48,12 @@ import { useOsReceiveApi } from '/app/view/OsReceive/hooks/useOsReceiveApi'
 import { useSecureBackgroundSplashScreen } from '/hooks/useSplashScreen'
 import { hideSplashScreen } from '/app/theme/SplashScreenService'
 
+import {
+  useLauncherContext,
+  LauncherContextProvider
+} from './screens/home/hooks/useLauncherContext'
+import LauncherView from './screens/konnectors/LauncherView'
```
Comment (Crash--) [inline src/App.js:55]: alias 🙏

Comment (doubleface) [reply src/App.js]: Fixed

File: __tests__/jest.config.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ const config = {
       '<rootDir>/__tests__/transformer/imageTransformer.js'
   },
   transformIgnorePatterns: [
-    'node_modules/(?!((jest-)?react-native(-.*)?|@react-native(-community)?|p-timeout?|p-wait-for?|@notifee?)/)'
+    'node_modules/(?!((jest-)?react-native(-.*)?|@react-native(-community)?|p-timeout?|p-wait-for?|@notifee?)|@fengweichong/react-native-gzip?/)'
```
Comment (Ldoppea) [inline __tests__/jest.config.js:16]: Why is this edit needed? Seems like this is not related with the PR, is it?

File: src/screens/home/hooks/useLauncherContext.tsx [hunk 3]
```diff
@@ -95,7 +96,11 @@ export const LauncherContextProvider: React.FC<Props> = ({
             <ErrorParallelKonnectors
               currentRunningKonnector={launcherContext.value.konnector.slug}
               concurrentKonnector={concurrentKonnector}
-              onClose={(): void => setConcurrentKonnector(undefined)}
+              onClose={(): void => {
+                setConcurrentKonnector(undefined)
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+                launcherEvent.emit('launchResult', { cancel: true })
```
Comment (Ldoppea) [inline src/screens/home/hooks/useLauncherContext.tsx:102]: I'm not sure to understand this one. How do we distinguish this `launchResult` from the one that would be emited by the "already running" connector?

File: src/components/webviews/CozyWebView.js [hunk 4]
```diff
@@ -136,6 +172,10 @@ export const CozyWebView = ({
     return () => BiometryEmitter.off('change', injectSettings)
   }, [injectSettings, webviewRef])
 
+  if (shouldShowCliskDevMode() && launcherContext.state === 'default') {
+    return <CliskDevView setLauncherContext={trySetLauncherContext} />
+  }
+
```
Comment (Ldoppea) [inline src/components/webviews/CozyWebView.js:178]: By doing this is the CozyWebView, this means that both homescreen and cozyappscreen will display the CliskDevView when `shouldShowCliskDevMode`. So it will be displayed twice. Is that expected/problematic?

File: src/components/webviews/CozyWebView.js [hunk 5]
```diff
@@ -182,6 +222,14 @@ export const CozyWebView = ({
         tryConsole(m, log, logId)
         nativeIntent?.tryEmit(m)
 
+        const data = m?.nativeEvent?.data
+        if (data) {
+          const { message, value } = JSON.parse(data)
+          if (message === 'startLauncher') {
+            trySetLauncherContext({ state: 'launch', value })
+          }
+        }
```
Comment (Ldoppea) [inline src/components/webviews/CozyWebView.js:231]: nit: It would be perfect if this is moved in an external method like it is done by `tryConsole(m,...), `tryNavigatorShare(m,...)` and `tryCrypto(m,...)`. But only if this is easy. Current implementation is good.

File: src/App.js [hunk 6]
```diff
@@ -76,11 +83,37 @@ const App = ({ setClient }) => {
...
+    <>
+      <RootNavigator initialRoute={initialRoute} setClient={setClient} />
+      {canDisplayLauncher() && (
+        <LauncherView
+          launcherClient={launcherClient}
+          launcherContext={launcherContext.value}
+          retry={resetLauncherContext}
+          setLauncherContext={setLauncherContext}
+          onKonnectorLog={onKonnectorLog}
+          onKonnectorJobUpdate={onKonnectorJobUpdate}
```
Comment (Ldoppea) [inline src/App.js:111]: Now that LauncherView is under LauncherContext, we can retrieve all those props directly in the component using `useLauncherContext`, so we don't need to inject them here.

File: src/components/webviews/CozyWebView.js (hunk 4 above)
Comment (Ldoppea) [reply src/components/webviews/CozyWebView.js:178]: PS: after some reflexion I indeed think that this may be problematic as we expect all our `CozyProxyWebView`, `CozyWebView`, `ReloadInterceptorWebView` (etc) to always return a WebView and to be interchangeable. So maybe we can keep this code in the HomeScreen where it was?

File: __tests__/jest.config.js (hunk 2 above)
Comment (doubleface) [reply __tests__/jest.config.js:16]: This is needed to run unit tests. Now that CozyWebView imports useLauncherContext, which ends up importing `src/libs/cozyAppBundle/cozyAppBundle.ts` which imports react-native-gzip.

File: src/screens/home/hooks/useLauncherContext.tsx (hunk 3 above)
Comment (doubleface) [reply src/screens/home/hooks/useLauncherContext.tsx:102]: Actually, there is no distinction. The current account/konnector opened in harvest will receive this message. And since the current opened connector is the one which caused to open the "already running" connector, this is the one we want to cancel.

File: src/components/webviews/CozyWebView.js (hunk 5 above)
Comment (doubleface) [reply src/components/webviews/CozyWebView.js]: Done

File: src/App.js (hunk 6 above)
Comment (doubleface) [reply src/App.js:111]: Yes but the LauncherView is a class component and I can't use hooks in it.

File: src/components/webviews/CozyWebView.js (hunk 4 above)
Comment (doubleface) [reply src/components/webviews/CozyWebView.js]: Right, this is done now

File: src/App.js (hunk 6 above)
Comment (doubleface) [reply src/App.js:111]: And changing the LaucherView to a function would be more difficult at the moment.

**Comment by doubleface** [comment]
@Crash-- @Ldoppea I think the PR is now ready

**Review by Ldoppea — APPROVED** [review]
